flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/PlanningConfigurationBuilder.java

```diff
@@ -155,11 +156,23 @@ public Context getContext() {
 			SqlParser
 				.configBuilder()
 				.setParserFactory(FlinkSqlParserImpl.FACTORY)
-				.setConformance(FlinkSqlConformance.DEFAULT)
+				.setConformance(getSqlConformance())
 				.setLex(Lex.JAVA)
 				.build());
 	}
 
+	private FlinkSqlConformance getSqlConformance() {
+		SqlDialect sqlDialect = tableConfig.getSqlDialect();
+		switch (sqlDialect) {
+		case HIVE:
+			return FlinkSqlConformance.HIVE;
+		case DEFAULT:
+			return FlinkSqlConformance.DEFAULT;
+		default:
```

Comment:
Do not add a default. Because otherwise people might forget to update this switch once a new dialect is added.